students/hirot_directory/fib_hiroTakechi.py

```diff
@@ -0,0 +1,53 @@
+
+'''
+Fibonacchi recursion Python code
+
+'''
+
+# in-class:
+
+def fib(n):
+
+	if n == 0:
+		return 0
+	if n == 1:
+		return 1
+	return fib(n-1) + fib(n-2)
+
+if __name__ == '__main__':
+	assert fib(0) == 0
+	assert fib(1) == 1
+	assert fib(2) == 1
+	assert fib(3) == 2
+	print(fib(10))
+
+# the codes that I worked on after class: 
+
+## code 1:
+def fibonacci(n, a=1, b=0):
+    return b if n < 1 else fibonacci(n - 1, a + b, a)
+
+print(fibonacci(7))
+
+
+## code 2:
+class Program():
```

Comment:
we'll get to classes in a couple weeks, but this is not an appropriate use case.

And it doesn't work -- the Fib method is calling a global Fib()function -- or it will fail if there isn't one.

Reply:
Hi Chris,

Thank you for the feedback. I just attended Maria's office hours and she also said we would cover the class and OOP in a few weeks. I just wanted to try it since Foundations of Python Programming class required to use their OOP template..

Thanks,
Hiro